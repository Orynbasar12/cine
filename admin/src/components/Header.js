import React, { Component } from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { Layout, Menu, Icon } from "antd";

import { logout as logoutAction } from '../actions/auth';

const menuItems = [
  { link: "/schools", name: "Schools", icon: "shop" },
  { link: "/users", name: "Users", icon: "user" },
  { link: "/faculties", name: "Faculties", icon: "table" },
  { link: "/courses", name: "Courses", icon: "database" },
  { link: "/chapters", name: "Chapters", icon: "profile" },
  { link: "/modules", name: "Modules", icon: "file" },
  { link: "/skills", name: "Skills", icon: "tool" },
  { link: "/notifications", name: "Notifications", icon: "notification" },
  { link: "/contests", name: "Contests", icon: "trophy" },
  { link: "/certificates", name: "Certificates", icon: "file-markdown" },
  { link: "/events", name: "Events", icon: "hourglass" },
  { link: "/tariffs", name: "Tariffs", icon: "dollar" }
];

class Header extends Component {
  state = {
    current: ""
  };

  handleClick = e => {
    this.setState({
      current: e.key
    });
  };

  renderContent() {
    if (this.props.auth) {
      return (
        <Menu
          onClick={this.handleClick}
          selectedKeys={[this.state.current]}
          mode="horizontal"
          theme="dark"
        >
          {this.props.auth.role === "admin" &&
            menuItems.map(item => (
              <Menu.Item key={item.name}>
                <Link to={item.link} key={item.link}>
                  <Icon type={item.icon} /> {item.name}
                </Link>
              </Menu.Item>
            ))}

          {this.props.auth.role === "reviewer" && (
            <Menu.Item key="submissions">
              <Link to="/submissions" key="submissions">
                <Icon type="file" /> Submissions
              </Link>
            </Menu.Item>
          )}

          <Menu.Item
            key="logout"
          >
            <a href="/" onClick={this.props.logout}>
              <Icon type="logout" /> Log Out
            </a>
          </Menu.Item>
        </Menu>
      );
    }
    return (
      <a href="/login">
        <Icon type="login" /> Login
      </a>
    );
  }

  render() {
    return (
      <Layout.Header style={{ height: 46 }}>
        {this.renderContent()}
      </Layout.Header>
    );
  }
}

Header.propTypes = {
  auth: PropTypes.shape({
    role: PropTypes.string
  }),
  logout: PropTypes.func.isRequired
};

Header.defaultProps = {
  auth: {}
};

function mapStateToProps({ auth }) {
  return { auth };
}

const mapDispatchToProps = (dispatch) => ({
    logout: () => logoutAction(dispatch)
  });

export default connect(mapStateToProps, mapDispatchToProps)(Header);
