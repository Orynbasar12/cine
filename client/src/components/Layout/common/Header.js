import React, { Component } from "react";
import { Link, withRouter } from "react-router-dom";
import { connect } from "react-redux";
import Modal from "react-modal";

import Dropdown from "./Dropdown";
import { logoutUser } from "../../../actions/authActions";
import { clearCurrentProfile } from "../../../actions/profileActions";
import { fileserver } from "../../../constants";

class Header extends Component {
  state = {
    text: "",
    dropdown: false
  };

  componentWillMount() {
    Modal.setAppElement("body");
  }

  componentDidMount() {
    switch (this.props.title) {
      case "/student/":
        return this.setState({ text: "Главная" });
      case "/student/universities":
        return this.setState({ text: "Университеты" });
      case "/student/help":
        return this.setState({ text: "Помощь" });
    }
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.location.pathname === "/student/") {
      this.setState({
        text: "Главная"
      });
    }
    if (nextProps.location.pathname === "/student/universities") {
      this.setState({
        text: "Университеты"
      });
    }
    if (nextProps.location.pathname === "/student/help") {
      this.setState({
        text: "Помощь"
      });
    }
  }

  showDropdown = () => {
    this.setState({
      dropdown: true
    });
  };

  closeDropdown = () => {
    this.setState({
      dropdown: false
    });
  };

  logoutUser = e => {
    e.preventDefault();
    this.props.clearCurrentProfile();
    this.props.logoutUser(this.props.history);
  };

  render() {
    const { isAuthenticated, user } = this.props.auth;
    if (this.state.text === "") {
      return <div />;
    }
    return (
      <header className="header">
        <div className="header-left-part">
          <span className="text-18">{this.state.text}</span>
          <div className="header-info">
            {this.props.auth.user.role === "student" ? (
              <button className="btn-app">
                <Link to="/student/apply/Выберите ВУЗ/0/Выберите факультет/0/Выберите специальность/0">
                  Подать заявку
                </Link>
              </button>
            ) : null}
            <div className="avatar">
              <img
                src={`http://${fileserver}:9999/${this.props.auth.user.avatar}`}
              />
            </div>
            <img
              onClick={this.showDropdown}
              className="dropdown"
              src="/images/dropdown.svg"
            />
          </div>
        </div>
        <Modal
          isOpen={this.state.dropdown}
          onRequestClose={this.closeDropdown}
          contentLabel="Example Modal"
          contentLabel="Example Modal"
          className="dropdown-menu"
          overlayClassName="overlay-err"
        >
          <Dropdown
            closeDropdown={this.closeDropdown}
            logoutUser={this.logoutUser}
          />
        </Modal>
      </header>
    );
  }
}

const mapStateToProps = state => ({
  auth: state.auth
});

export default connect(
  mapStateToProps,
  { logoutUser, clearCurrentProfile }
)(withRouter(Header));
