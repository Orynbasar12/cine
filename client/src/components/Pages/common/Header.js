import React from 'react'

const Header = props =>

  <header className='header header-w'>
    <span className="header-w-text">Всего {props.count} университетов</span>
  </header>

export default Header
