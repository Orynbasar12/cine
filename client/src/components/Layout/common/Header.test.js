import { shallow } from 'enzyme'
import React from 'react'

import Header from './Header'

it('expect header component', () => {
  expect(shallow(<Header />)).toMatchSnapshot()
})
