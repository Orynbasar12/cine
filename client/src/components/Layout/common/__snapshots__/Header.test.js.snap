// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect header component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="header-left-part"
      >
        <span>
          Главная
        </span>
        <div
          className="header-info"
        >
          <button
            className="btn-app"
          >
            <Link
              replace={false}
              to="/profile/apply"
            >
              Подать заявку
            </Link>
          </button>
          <div
            className="avatar"
          >
            <img
              src="/images/avatar.png"
            />
          </div>
          <img
            className="dropdown"
            src="/images/dropdown.svg"
          />
        </div>
      </div>,
      "className": "header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            Главная
          </span>,
          <div
            className="header-info"
          >
            <button
              className="btn-app"
            >
              <Link
                replace={false}
                to="/profile/apply"
              >
                Подать заявку
              </Link>
            </button>
            <div
              className="avatar"
            >
              <img
                src="/images/avatar.png"
              />
            </div>
            <img
              className="dropdown"
              src="/images/dropdown.svg"
            />
          </div>,
        ],
        "className": "header-left-part",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Главная",
          },
          "ref": null,
          "rendered": "Главная",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="btn-app"
              >
                <Link
                  replace={false}
                  to="/profile/apply"
                >
                  Подать заявку
                </Link>
              </button>,
              <div
                className="avatar"
              >
                <img
                  src="/images/avatar.png"
                />
              </div>,
              <img
                className="dropdown"
                src="/images/dropdown.svg"
              />,
            ],
            "className": "header-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/profile/apply"
                >
                  Подать заявку
                </Link>,
                "className": "btn-app",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Подать заявку",
                  "replace": false,
                  "to": "/profile/apply",
                },
                "ref": null,
                "rendered": "Подать заявку",
                "type": [Function],
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  src="/images/avatar.png"
                />,
                "className": "avatar",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "src": "/images/avatar.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "dropdown",
                "src": "/images/dropdown.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="header-left-part"
        >
          <span>
            Главная
          </span>
          <div
            className="header-info"
          >
            <button
              className="btn-app"
            >
              <Link
                replace={false}
                to="/profile/apply"
              >
                Подать заявку
              </Link>
            </button>
            <div
              className="avatar"
            >
              <img
                src="/images/avatar.png"
              />
            </div>
            <img
              className="dropdown"
              src="/images/dropdown.svg"
            />
          </div>
        </div>,
        "className": "header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              Главная
            </span>,
            <div
              className="header-info"
            >
              <button
                className="btn-app"
              >
                <Link
                  replace={false}
                  to="/profile/apply"
                >
                  Подать заявку
                </Link>
              </button>
              <div
                className="avatar"
              >
                <img
                  src="/images/avatar.png"
                />
              </div>
              <img
                className="dropdown"
                src="/images/dropdown.svg"
              />
            </div>,
          ],
          "className": "header-left-part",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Главная",
            },
            "ref": null,
            "rendered": "Главная",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="btn-app"
                >
                  <Link
                    replace={false}
                    to="/profile/apply"
                  >
                    Подать заявку
                  </Link>
                </button>,
                <div
                  className="avatar"
                >
                  <img
                    src="/images/avatar.png"
                  />
                </div>,
                <img
                  className="dropdown"
                  src="/images/dropdown.svg"
                />,
              ],
              "className": "header-info",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/profile/apply"
                  >
                    Подать заявку
                  </Link>,
                  "className": "btn-app",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Подать заявку",
                    "replace": false,
                    "to": "/profile/apply",
                  },
                  "ref": null,
                  "rendered": "Подать заявку",
                  "type": [Function],
                },
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    src="/images/avatar.png"
                  />,
                  "className": "avatar",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "src": "/images/avatar.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "dropdown",
                  "src": "/images/dropdown.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
